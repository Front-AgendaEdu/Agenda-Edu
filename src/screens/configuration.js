import { StatusBar } from 'expo-status-bar'
import React from 'react'
import Expo from "expo"
import { StyleSheet, Text, View, ActivityIndicator, Image } from 'react-native'
import { useNavigation, CommonActions } from '@react-navigation/native'
import Styles from '../styles/styles'
import AsyncStorage from '@react-native-community/async-storage'
import * as Font from 'expo-font'

const auth = require('../functions/auth')
const styled = require('../styledComponets/styles')

class Configuration extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            label: "Carregando...",
            loading: true,
        }
    }

    async componentDidMount() {

        await Font.loadAsync({
            'SF-Pro-Display-Ultralight': require('../assets/fonts/SF-Pro-Display-Ultralight.ttf'),
            'SFProDisplay-Regular': require('../assets/fonts/SFProDisplay-Regular.ttf'),
            'SFProDisplay-Bold': require('../assets/fonts/SFProDisplay-Bold.ttf'),
            'SFProDisplay-Semibold': require('../assets/fonts/SFProDisplay-Semibold.ttf'),
            'SFProDisplay-Medium': require('../assets/fonts/SFProDisplay-Medium.ttf'),
        })
        this.setState({ loading: false });

        const { navigation } = this.props
        let email = await AsyncStorage.getItem('EMAIL')
        let psw = await AsyncStorage.getItem('PSW')

        try {
            navigation.dispatch(
                CommonActions.reset({
                    index: 0,
                    routes: [{ name: await auth.login(email, psw) ? 'Events' : 'Login' }],
                }))
        } catch (error) {
            console.log(error.toString(error));
        }
    }

    render() {
        if (this.state.loading) {
            return <ActivityIndicator />
        }
        return (
            <styled.configContainer>
                <styled.configIcon source={require('../assets/imgs/icon.png')}/>
                <styled.txtLoading>{this.state.label}</styled.txtLoading>
                <StatusBar style='auto' />
            </styled.configContainer>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#733DBE',
        alignItems: 'center',
        justifyContent: 'center'
    }
})


export default function ({ props, route }) {
    const navigation = useNavigation()

    return <Configuration {...props} navigation={navigation} />
}
