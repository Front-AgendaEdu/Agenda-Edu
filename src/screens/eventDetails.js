import { StatusBar } from 'expo-status-bar'
import React from 'react'
import { StyleSheet, Text, View, TextInput, Dimensions, KeyboardAvoidingView, TouchableOpacity, ImageBackground, ScrollView } from 'react-native'
import { useNavigation, CommonActions } from '@react-navigation/native'
import Styles from '../styles/styles'
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons'
import EvilIcons from 'react-native-vector-icons/EvilIcons'
import * as Font from 'expo-font'
import { AppLoading } from 'expo'

const screenHeight = Math.round(Dimensions.get('window').height)
const screenWidth = Math.round(Dimensions.get('window').width)

const widthPercent = (screenWidth / 100)
const heightPercent = (screenHeight / 100)

const auth = require('../functions/auth')

class EventDetails extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            email: 'student@ae.com',
            psw: '',
            isVisiblePsw: false,
            label: "Login",
            borderColorEmail: '#ABB1B7',
            borderColorPsw: '#ABB1B7',
        }
    }

    async componentDidMount() {

    }

    render() {
        const { navigation, item } = this.props


        return (
            <View style={Styles.container}>
                <ImageBackground source={require('../assets/imgs/image.jpeg')} style={Styles.eventDetailsBackground}>
                    <View style={Styles.containerDetails}>
                        <View style={Styles.headersDetailsView}>
                            <View style={Styles.dateView}>

                                <View style={Styles.txtDayMonthView}>
                                    <Text style={Styles.txtDayMonth}>{item.sendAt.getDate()}</Text>
                                    <Text style={Styles.txtMonth}>{ item.curencyMonth.substring(0, 3).toUpperCase()}</Text>
                                </View>
                                {/* <View style={Styles.txtMonthView}>
                                    <Text style={Styles.txtMonth}>JAN</Text>
                                </View> */}
                            </View>
                            <View style={Styles.eventDetailsTitleView}>
                                <Text style={Styles.eventDetailsTitle}>{item.title}</Text>
                            </View>
                        </View>

                        <View style={Styles.headersClockDetailsView}>
                            <EvilIcons
                                style={Styles.eventDetailsClock}
                                size={widthPercent * 6.5}
                                color="#666666"
                                containerStyle={{ backgroundColor: "#F00" }}
                                name={'clock'}
                            />
                            {/* <Text style={Styles.txtClock}>{`${item.sendAt.getHours() < 10 ? `0${item.sendAt.getHours()}` : item.sendAt.getHours()}:00`}</Text> */}
                            <Text style={Styles.txtClock}>{item.editedTimeSendAt}</Text>

                        </View>

                        <View style={Styles.descriptionView}>
                            <ScrollView>
                                <Text style={Styles.txtEventDescription}>{item.description}</Text>
                            </ScrollView>
                        </View>

                    </View>
                </ImageBackground>
                <StatusBar style='auto' />
            </View >
        )
    }
}

export default function ({ props, route }) {
    const navigation = useNavigation()
    const { item } = route.params

    return <EventDetails {...props} navigation={navigation} item={item} />
}



