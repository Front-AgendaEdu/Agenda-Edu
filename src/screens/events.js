import { StatusBar } from 'expo-status-bar'
import React from 'react'
import { StyleSheet, Text, View, TouchableOpacity, Dimensions, Image, LogBox, Animated, Alert, ActivityIndicator, FlatList } from 'react-native'
import { useNavigation, CommonActions } from '@react-navigation/native'
import AsyncStorage from '@react-native-community/async-storage'
import Styles from '../styles/styles'
import { Drawer } from 'react-native-material-drawer'
import EvilIcons from 'react-native-vector-icons/EvilIcons'
import { Icon } from 'react-native-elements'


const auth = require('../functions/auth')

const screenWidth = Dimensions.get('window').width
const screenHeight = Dimensions.get('window').height

const widthPercent = (screenWidth / 100)
const heightPercent = (screenHeight / 100)

LogBox.ignoreLogs(['Warning: ...']);
LogBox.ignoreAllLogs();

var dataComponent = []
var metaDataComponent = []

var options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' }

options.timeZone = 'UTC';
options.timeZoneName = 'short';

const curencyMonth = ["janeiro", "fevereiro", "março", "Abril", "Maio", "Junho", "Julho", "agosto", "setempro", "outubro", "novembro", "dezembro"]
const weekday = ["Segunda", "Terça", "Quarta", "Quinta", "Sexta", "Sábado", "Domingo"]

class Home extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			label: "Eventos",
			isOpen: false,
			limit: 3,
			page: 1,
			data: null,
			metadata: null,
			dataLoad: false,
			scrollY: new Animated.Value(0),
			loadMore: true,
			dataComponent: [],
			total_pages: 1,
		}

	}

	async getEvents() {
		const myRequest = new Request(
			`https://frontend-test.agendaedu.com/api/events?limit=${this.state.limit};page=${this.state.page}`,
			{
				method: 'GET',
				headers: {
					token: await AsyncStorage.getItem('AUTH_TOKEN'),
					Accept: 'application/json', 'Content-Type': 'application/json'
				}
			}
		)

		fetch(myRequest)
			.then(response => {
				if (response.status === 200) {
					return response.json()
				} else {
					throw new Error('erro em no servidor.')
				}
			})
			.then(response => {
				response.data.forEach(function (item, indice) {
					let sendAt = new Date(item.sendAt)
					let startAt = new Date(item.startAt)

					let editedTimeSendAt = `${sendAt.getUTCHours() < 10 ? `0${sendAt.getUTCHours()}` : sendAt.getUTCHours()}:00`
					let editedTimeStartAt = `${startAt.getUTCHours() < 10 ? `0${startAt.getUTCHours()}` : startAt.getUTCHours()}:00`

					dataComponent.push({
						title: item.title,
						image: item.image,
						sendAt: sendAt,
						startAt: startAt,
						image: item.image,
						description: item.description,
						editedTimeSendAt: editedTimeSendAt,
						editedDataStartAt: editedTimeStartAt,
						editedDataTimeStartAt: `${weekday[startAt.getUTCDay()]}, ${startAt.getUTCDate()} de ${curencyMonth[startAt.getUTCMonth()]} às ${editedTimeStartAt}h`,
						weekday: weekday[sendAt.getUTCDay()],
						curencyMonth: curencyMonth[sendAt.getUTCMonth()]
					})
				});

				 this.setState({total_pages: response.metadata.total_pages}) 

	
				dataComponent.sort(function (a, b) {
					return new Date(a.sendAt).getTime() > new Date(b.sendAt).getTime() ? -1 : new Date(a.sendAt).getTime() < new Date(b.sendAt).getTime() ? 1 : 0
				})

				this.setState({ dataComponent: dataComponent, dataLoad: true, loadMore: true })

			})
			.catch(error => {
				console.log(`RESPONSE: ${JSON.stringify(error)} `)
			})
	}


	async componentDidMount() {
		const { navigation } = this.props

		navigation.setOptions({
			headerLeft: () => (
				<TouchableOpacity onPress={() => { this.setState({ isOpen: !this.state.isOpen }) }}>
					<Icon
						size={widthPercent * 8}
						color="#000"
						containerStyle={{ marginLeft: widthPercent * 4, backgroundColor: "#FFF" }}
						type="ionicon"
						name={Platform.OS === 'ios' ? 'ios-menu' : 'md-menu'}
					/>
				</TouchableOpacity>
			),
		})

		await this.getEvents()
	}

	_scrollView(f) {
		f.getScrollResponder().scrollTo({ x: 140, y: 140, animated: false })
	}

	handleScroll(event) {
		console.log(event.nativeEvent.contentOffset.y);
	}

	loadMoreData = async () => {
		const { loadMore } = this.state
		if (loadMore) {
			await this.setState({ loadMore: false, page: this.state.page < this.state.total_pages ? this.state.page + 1 : this.state.page })

			if (this.state.page < this.state.total_pages) {
				await this.getEvents()
			} else {
				this.setState({ loadMore: true })
			}

			return
		}
	}

	isCloseToBottom = ({ layoutMeasurement, contentOffset, contentSize }) => {
		const paddingToBottom = 20
		return layoutMeasurement.height + contentOffset.y >=
			contentSize.height - paddingToBottom
	}

	render() {
		const { navigation } = this.props

		return (
			<View style={styles.container}>
				<Drawer
					style={{ justifyContent: "center" }}
					width={widthPercent * 70}
					open={this.state.isOpen}
					drawerContent={
						<View style={Styles.lateralMenu}>
							<TouchableOpacity
								testID={'eventCardButton'}
								style={Styles.btnSair}
								onPress={() => { this.setState({ page: 0 }), dataComponent = [], auth.logOut(navigation) }}
							>
								<Text style={Styles.labelBtnEntrar}>Sair</Text>
							</TouchableOpacity>
						</View>
					}
					onClose={() => this.setState({ isOpen: false })}
					animationTime={250}
				>

						<FlatList
							contentContainerStyle={{alignItems: 'center', paddingBottom: widthPercent * 5}}
							ref={(view) => this._scrollView = view}
							scrollEventThrottle={16}
							onScroll={Animated.event(
								[{ nativeEvent: { contentOffset: { y: this.state.scrollY } } }],
								{
									listener: event => {
										if (this.isCloseToBottom(event.nativeEvent)) {
											this.loadMoreData()
										}
									}
								}
							)}
							data={this.state.dataComponent}
							renderItem={({ item }) =>

								<View style={Styles.mainContainer}>
									<View style={Styles.header}>
										<View style={{ flex: 1.5 }}>
											<Text style={Styles.headerLabel}>{`${item.weekday}, ${item.sendAt.getUTCDate()} de ${item.curencyMonth}`}</Text>
										</View>
										<View style={{ flex: 1.2 }}>
											<View style={Styles.headerLine} />
										</View>
									</View>

									<TouchableOpacity style={Styles.eventContainer}
										onPress={() => { navigation.navigate("EventDetails", { item: item }) }}
									>

										<View style={Styles.eventLateralMarker} />
										{item.image != null ?
											<View style={Styles.imageView}>
												<Image
													style={Styles.cardImage}
													source={{
														uri: item.image,
													}}
												/>
											</View>
											:
											null
										}

										<View style={Styles.eventContent} >
											<Text style={Styles.eventContentType}>EVENTOS</Text>
											<Text ellipsizeMode='tail' numberOfLines={1} style={Styles.eventTitle}>{`${item.title}`}</Text>
											<View style={Styles.clockView}>
												<EvilIcons
													size={widthPercent * 6.5}
													color="#666666"
													containerStyle={{ backgroundColor: "#F00" }}
													name={'clock'}
												/>
												<Text style={Styles.txtClock}>{item.editedTimeSendAt}</Text>
											</View>
											<Text style={Styles.eventDateTime}>{item.editedDataTimeStartAt}</Text>
										</View>
									</TouchableOpacity>
								</View>
							}
						/>
					{!this.state.loadMore
						?
						<View style={Styles.loadView}>
							<Text style={Styles.txtLoadingEvents}>{`Carregando...`}</Text>
						</View>
						:
						null
					}

				</Drawer>

				<StatusBar style='auto' />
			</View>
		)
	}
}


const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#fff',
		alignItems: 'center',
		justifyContent: 'center'
	},
	containerLoading: {
		flex: 1,
		justifyContent: 'center',
	},
	horizontal: {
		flexDirection: 'row',
		justifyContent: 'space-around',
		padding: 10,
	},
})



export default function ({ props, route }) {
	const navigation = useNavigation()

	return <Home {...props} navigation={navigation} />
}
