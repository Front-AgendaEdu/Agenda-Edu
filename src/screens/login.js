import { StatusBar } from 'expo-status-bar'
import React from 'react'
import { StyleSheet, Text, View, TextInput, Dimensions, KeyboardAvoidingView, TouchableOpacity, Keyboard, TouchableWithoutFeedback, Platform } from 'react-native'
import { useNavigation, CommonActions } from '@react-navigation/native'
import Styles from '../styles/styles'
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons'
// import StyledComponents from '../styled-componets/styles'
// import {Container ,TextInputLabel} from '../styledComponets/styles'


const screenHeight = Math.round(Dimensions.get('window').height)
const screenWidth = Math.round(Dimensions.get('window').width)

const widthPercent = (screenWidth / 100)
const heightPercent = (screenHeight / 100)

const DismissKeyboard = ({ children }) => (
    <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
        {children}
    </TouchableWithoutFeedback>
);


global.isVisiblePsw = false
const auth = require('../functions/auth')
const styled = require('../styledComponets/styles')

export class Login extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            email: 'student@ae.com',
            psw: '',
            isVisiblePsw: false,
            label: "Login",
            borderColorEmail: '#ABB1B7',
            borderColorPsw: '#ABB1B7',
        }
    }

    async componentDidMount() {

    }

    render() {
        const { navigation } = this.props

        return (
            <DismissKeyboard>
                <KeyboardAvoidingView style={Styles.container} behavior="padding" enabled={Platform.OS === 'ios' ? true : false} >
                    <styled.Container>
                        <styled.containerLoginInput>
                            <styled.LoginTitle>
                                FaÃ§a seu login ðŸ”‘
                            </styled.LoginTitle>

                            <styled.TextInputLabel>
                                Email ou usuÃ¡rio
                            </styled.TextInputLabel>

                            <styled.InputView>
                                <styled.EmailInput
                                    focusColor={this.state.borderColorEmail}
                                    testID={'inputEmail'}
                                    value={this.state.email}
                                    onFocus={() => { this.setState({ borderColorEmail: '#733DBE' }), global.inputEmail = true }}
                                    onBlur={() => { this.setState({ borderColorEmail: '#ABB1B7' }) }}
                                    autoCapitalize={'none'}
                                    onChangeText={value => { this.setState({ email: value }) }}
                                    placeholderTextColor={'rgba(0, 0, 0, 0.5)'}
                                    underlineColorAndroid='transparent'
                                    keyboardType={'email-address'}

                                />
                                <styled.EmailInputIcon
                                    name={'email-outline'}
                                />
                            </styled.InputView>
                        </styled.containerLoginInput>

                        <styled.containerPsw >
                            <styled.TextInputLabel>
                                Senha
                            </styled.TextInputLabel>

                            <styled.InputView>
                                <styled.PswlInput
                                    testID={'inputPassword'}
                                    focusColor={this.state.borderColorPsw}
                                    value={this.state.psw}
                                    onFocus={() => { this.setState({ borderColorPsw: '#733DBE' }), global.inputPassword = true }}
                                    onBlur={() => { this.setState({ borderColorPsw: '#ABB1B7' }) }}
                                    autoCapitalize={'none'}
                                    onChangeText={value => { this.setState({ psw: value }) }}
                                    placeholderTextColor={'rgba(0, 0, 0, 0.5)'}
                                    underlineColorAndroid='transparent'
                                    secureTextEntry={!this.state.isVisiblePsw}
                                />
                                <styled.BtnView
                                    testID={'showPswBtn'}
                                    onPress={() => { this.setState({ isVisiblePsw: !this.state.isVisiblePsw }), global.isVisiblePsw = !this.state.isVisiblePsw }}
                                >
                                    <styled.PswlInputIcon
                                        name={this.state.isVisiblePsw == true ? 'eye-off' : 'eye'}
                                    />
                                </styled.BtnView>
                            </styled.InputView>
                        </styled.containerPsw>
                    </styled.Container>


                    <styled.btnEntrarContainer>
                        <styled.btnEntrar
                            onPress={() => { auth.login(this.state.email, this.state.psw, true, navigation) }}
                            testID={'entryButton'}

                        >
                            <styled.labelBtn>Entrar</styled.labelBtn>
                        </styled.btnEntrar>
                        <StatusBar style='auto' />
                    </styled.btnEntrarContainer>

                </KeyboardAvoidingView>
            </DismissKeyboard>
        )
    }
}

export default function ({ props, route }) {
    const navigation = useNavigation()

    return <Login {...props} navigation={navigation} />
}
