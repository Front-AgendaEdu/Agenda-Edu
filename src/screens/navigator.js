/**

 */

import React, { Component } from 'react'
import { SafeAreaView, StyleSheet, ScrollView, View, Text, StatusBar, Button, TouchableOpacity, Dimensions } from 'react-native'

import { NavigationContainer, StackActions, getFocusedRouteNameFromRoute } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'

// import AsyncStorage from '@react-native-community/async-storage'

import Login from './login'
import Configuration from './configuration'
import Events from './events'
import EventDetails from './eventDetails'


const Stack = createStackNavigator();

class Navigator extends React.Component {

  constructor(props) {
    super(props)
    this.state = {

    }
  }

  render() {

    return (
      <NavigationContainer>
        <Stack.Navigator
          screenOptions={{
            title: '',
            headerShown: false,
            headerStyle: {
              backgroundColor: '#FFF',
            },
            headerTintColor: '#fff',
            headerTitleStyle: {
              // fontWeight: 'bold',
            },
          }}
        >
          {/* <Stack.Screen name="Configuration" component={Configuration} /> */}
          <Stack.Screen name="Configuration" component={Configuration}
            options={() => ({
              headerShown: false,
            })} />

          <Stack.Screen name="Login" component={Login}
            options={() => ({
              title: 'Login',
              headerShown: false,
              headerTintColor: '#000',
            })}
          />
          <Stack.Screen name="Events" component={Events}
            options={() => ({
              title: 'Eventos',
              headerShown: true,
              headerTintColor: '#000',
              backgroundColor: '#FEFEFE',
              headerTitleStyle: {
                marginLeft: 0,
                fontFamily: 'SFProDisplay-Medium',
              }
            })}
          />
          <Stack.Screen name="EventDetails" component={EventDetails}
            options={() => ({
              title: 'Detalhes do Evento',
              headerShown: true,
              headerTintColor: '#000',
              backgroundColor: '#FEFEFE',
              headerTitleStyle: {
                fontFamily: 'SFProDisplay-Medium',
                marginLeft: 0,
              }
            })}
          />

          {/* <Stack.Screen name="Main" component={Main}
            options={({ route, navigation }) => ({
              // headerTitle: 'Home Page',
              route: { route },
              navigation: { navigation },

            })}
          /> */}

        </Stack.Navigator>
      </NavigationContainer>

    )
  }

}

export default Navigator;


