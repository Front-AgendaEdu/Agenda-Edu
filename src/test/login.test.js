const mockedNavigate = jest.fn();

jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon')

jest.mock("@react-native-community/async-storage", () =>
	require("@react-native-community/async-storage/jest/async-storage-mock"),
);

jest.mock('@react-navigation/native', () => {
	return {
		...jest.requireActual('@react-navigation/native'),
		useNavigation: () => ({
			navigate: mockedNavigate,
		}),
	};
});


import React from 'react';
import { create, act } from 'react-test-renderer'
import { Login } from '../screens/login'
import renderer from 'react-test-renderer'


const LoginComponent = create(<Login />)

describe('##SNAPSHOT##', () => {
	it('Login Snapshot', async () => {
		const snap = renderer.create(<Login />).toJSON()

		expect(snap).toMatchSnapshot()
	})
})

describe('##ACTIONS CLICK##', () => {

	it('Verify if isVisiblePsw button status changed', () => {
		const button = LoginComponent.root.findByProps({ testID: 'showPswBtn' }).props
		act(() => button.onPress())
		expect(global.isVisiblePsw).toEqual(true);
	})

	it('Verify if entryButton click succesed', () => {
		const button = LoginComponent.root.findByProps({ testID: 'entryButton' }).props
		act(() => button.onPress())
		expect(global.callLoginTest).toEqual(true);
	})


	it('Verify if inputEmail focus succesed', () => {
		const inputEmail = LoginComponent.root.findByProps({ testID: 'inputEmail' }).props
		act(() => inputEmail.onFocus())
		expect(global.inputEmail).toEqual(true);
	})

	it('Verify if inputPassword focus succesed', () => {
		const inputPassword = LoginComponent.root.findByProps({ testID: 'inputPassword' }).props
		act(() => inputPassword.onFocus())
		expect(global.inputPassword).toEqual(true);
	})

})