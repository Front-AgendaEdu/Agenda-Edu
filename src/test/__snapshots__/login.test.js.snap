// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`##SNAPSHOT## Login Snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "flex": 1,
        "flexDirection": "column",
        "height": null,
        "justifyContent": "center",
        "width": null,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 54,
              "marginBottom": 60,
            },
          ]
        }
      >
        FaÃ§a seu login ðŸ”‘
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#666666",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 31.5,
            },
          ]
        }
      >
        Email ou usuÃ¡rio
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          focusColor="#ABB1B7"
          keyboardType="email-address"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholderTextColor="rgba(0, 0, 0, 0.5)"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "borderColor": "#ABB1B7",
                "borderRadius": 7.5,
                "borderStyle": "solid",
                "borderWidth": 1,
                "color": "#333333",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 37.5,
                "height": 126.73,
                "paddingLeft": "7%",
                "paddingRight": "14%",
                "width": 654,
              },
            ]
          }
          testID="inputEmail"
          underlineColorAndroid="transparent"
          value="student@ae.com"
        />
        <Icon
          name="email-outline"
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "color": "#AAAAAA",
                "fontSize": 60,
                "justifyContent": "center",
                "position": "absolute",
                "right": 22.5,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 26.68,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#666666",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 31.5,
            },
          ]
        }
      >
        Senha
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          focusColor="#ABB1B7"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholderTextColor="rgba(0, 0, 0, 0.5)"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Array [
              Object {
                "borderColor": "#ABB1B7",
                "borderRadius": 7.5,
                "borderStyle": "solid",
                "borderWidth": 1,
                "color": "#333333",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 37.5,
                "height": 126.73,
                "paddingLeft": "7%",
                "paddingRight": "14%",
                "width": 654,
              },
            ]
          }
          testID="inputPassword"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "justifyContent": "center",
              "opacity": 1,
              "position": "absolute",
              "right": 22.5,
            }
          }
          testID="showPswBtn"
        >
          <Icon
            name="eye"
            style={
              Array [
                Object {
                  "color": "#AAAAAA",
                  "fontSize": 60,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#733DBE",
          "borderRadius": 15,
          "bottom": 0,
          "height": 112.5,
          "justifyContent": "center",
          "marginBottom": "7%",
          "opacity": 1,
          "width": "82%",
        }
      }
      testID="entryButton"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "SFProDisplay-Semibold",
              "fontSize": 37.5,
            },
          ]
        }
      >
        Entrar
      </Text>
    </View>
  </View>
</View>
`;
